registry/service.go

```diff
@@ -52,11 +53,12 @@ func (s *Service) Auth(job *engine.Job) engine.Status {
 			if err != nil {
 				return job.Error(err)
 			}
+			secure = endpoint.secure
```

Comment:
Mmm... TBH I feel like I had to rebase for [this](https://github.com/docker/docker/commit/568f86eb186731b907b659e4ec64bda21c2fe31d#diff-2b07b8a4e93be324f246494ec743cde2R50), which you are now basically asking me to revert? :-)
You're more familiar with this piece of code than I am @dkjer, but is that your last word?